### PR TITLE
Domestic Payments

### PR DESCRIPTION
**Pull request by mariantiris**
Issue: https://github.com/SecureApiGateway/SecureApiGateway/issues/1555

Adding datamodel support for domestic payments v4.0.0

### CODE REVIEW

File: secure-api-gateway-ob-uk-common-obie-datamodel/src/main/java/uk/org/openbanking/datamodel/v4/payment/OBWriteDomesticConsentResponse5Data.java
Comment (dbadham-fr) [inline secure-api-gateway-ob-uk-common-obie-datamodel/src/main/java/uk/org/openbanking/datamodel/v4/payment/OBWriteDomesticConsentResponse5Data.java:32]: can you fix the import order please?

When you generate make sure you run the Intellij optimise imports on the code produced.

**Review by dbadham-fr — COMMENTED** [review]
can you add a description to the PR please?

File: secure-api-gateway-ob-uk-common-obie-datamodel/src/main/java/uk/org/openbanking/datamodel/v4/payment/OBWriteDomesticConsentResponse5Data.java
Comment (mariantiris) [reply secure-api-gateway-ob-uk-common-obie-datamodel/src/main/java/uk/org/openbanking/datamodel/v4/payment/OBWriteDomesticConsentResponse5Data.java:32]: Ran it multiple times to the point where this is not changing anymore. This is the order the IDE keeps it now.

Comment (dbadham-fr) [reply secure-api-gateway-ob-uk-common-obie-datamodel/src/main/java/uk/org/openbanking/datamodel/v4/payment/OBWriteDomesticConsentResponse5Data.java:32]: have you got the ForgeRock code style configured?

Comment (BohoCode) [reply secure-api-gateway-ob-uk-common-obie-datamodel/src/main/java/uk/org/openbanking/datamodel/v4/payment/OBWriteDomesticConsentResponse5Data.java:32]: [ForgeRock-IG.zip](https://github.com/user-attachments/files/17519457/ForgeRock-IG.zip)

Comment (BohoCode) [reply secure-api-gateway-ob-uk-common-obie-datamodel/src/main/java/uk/org/openbanking/datamodel/v4/payment/OBWriteDomesticConsentResponse5Data.java:32]: That can be imported to intellij's Settings->Editor->Code Style panel

Comment (BohoCode) [reply secure-api-gateway-ob-uk-common-obie-datamodel/src/main/java/uk/org/openbanking/datamodel/v4/payment/OBWriteDomesticConsentResponse5Data.java:32]: Here are the broader ForgeRock settings that work pretty well;
Instructions:
https://www.jetbrains.com/help/idea/configuring-code-style.html
[ForgeRock (1).zip](https://github.com/user-attachments/files/17519552/ForgeRock.1.zip)

Comment (mariantiris) [reply secure-api-gateway-ob-uk-common-obie-datamodel/src/main/java/uk/org/openbanking/datamodel/v4/payment/OBWriteDomesticConsentResponse5Data.java]: Thanks guys, applied the codestyles and reran optimize

Comment (dbadham-fr) [reply secure-api-gateway-ob-uk-common-obie-datamodel/src/main/java/uk/org/openbanking/datamodel/v4/payment/OBWriteDomesticConsentResponse5Data.java]: Thanks, PR looks good.

I'll hold off approving so we can keep the PR open to test out the new development workflow that we've been discussing.

**Review by dbadham-fr — APPROVED** [review]

**Pull request merged**
